Source/GSXib5KeyedUnarchiver.m

```diff
@@ -3005,6 +2962,79 @@ - (id) decodeObjectForXib: (GSXibElement*)element
   return object;
 }
 
+- (id) decodeSpecialButtonCellForKey: (NSString *)key
+{
+  // Search field encoding is real basic now...does not include these by default...
+  // So we're going to generate them here for now...again should be moved into
+  // class initWithCoder method eventually...
+  id object = AUTORELEASE([NSButtonCell new]);
+
+  unsigned int      bFlags = 0x8444000;
+  GSButtonCellFlags buttonCellFlags;
+
+  memcpy((void *)&buttonCellFlags,(void *)&bFlags,sizeof(struct _GSButtonCellFlags));
+
+  if ([@"NSSearchButtonCell" isEqualToString: key])
+    [object setTitle: @"search"];
+  else
+    [object setTitle: @"clear"];
+
+  [object setTransparent: buttonCellFlags.isTransparent];
+  [object setBordered: buttonCellFlags.isBordered];
+
+  [object setCellAttribute: NSPushInCell to: buttonCellFlags.isPushin];
+  [object setCellAttribute: NSCellLightsByBackground to: buttonCellFlags.highlightByBackground];
+  [object setCellAttribute: NSCellLightsByContents to: buttonCellFlags.highlightByContents];
+  [object setCellAttribute: NSCellLightsByGray to: buttonCellFlags.highlightByGray];
+  [object setCellAttribute: NSChangeBackgroundCell to: buttonCellFlags.changeBackground];
+  [object setCellAttribute: NSCellChangesContents to: buttonCellFlags.changeContents];
+  [object setCellAttribute: NSChangeGrayCell to: buttonCellFlags.changeGray];
+
+  if (buttonCellFlags.imageDoesOverlap)
+    {
+      if (buttonCellFlags.isImageAndText)
+        [object setImagePosition: NSImageOverlaps];
+      else
+        [object setImagePosition: NSImageOnly];
+    }
+  else if (buttonCellFlags.isImageAndText)
+    {
+      if (buttonCellFlags.isHorizontal)
+        {
+          if (buttonCellFlags.isBottomOrLeft)
+            [object setImagePosition: NSImageLeft];
+          else
+            [object setImagePosition: NSImageRight];
+        }
+      else
+        {
+          if (buttonCellFlags.isBottomOrLeft)
+            [object setImagePosition: NSImageBelow];
+          else
+            [object setImagePosition: NSImageAbove];
+        }
+    }
+  else
+    {
+      [object setImagePosition: NSNoImage];
+    }
+#if 0
```

Comment:
What's this?

```
#if 0
  [object setBordered: NO];
...
...
```

(I see this was in old code too, just wondering about it now)

Reply:
Yes, one of the cases where I keep the commented out code because it may be useful in the future. I really don't like that myself. Not sure what to do.